### ISSUE TITLE
Walking distances are off

### ISSUE THREAD

**Issue by thohug** [datamapio/posmo_one_testing#31]
**Setting**
Zurich, Switzerland, Android

wifi-fence: 1
do_not_track: 0
wifi_over_gps: 1
data_over_wifi_only: 1

**Issue**
Walking distances are wrong and differ strongly from reality (probably due to GPS inaccuracies).

9mins for 1km is quite fast:
![grafik](https://user-images.githubusercontent.com/67195528/102582896-d89ff100-4103-11eb-9e1c-329ff616d778.png)

Distance from google maps is about half of the distance measured: 
![grafik](https://user-images.githubusercontent.com/67195528/102582948-fa00dd00-4103-11eb-9fc7-ea0fd28d576e.png)

**Comment by rogerfischer** [comment]
This is about the distance, right? I think that wrong distance is definitely due to GPS noise. 
Only a graph based solution can correct this.

<img width="1277" alt="Screen Shot 2020-12-18 at 9 30 53 PM" src="https://user-images.githubusercontent.com/7697124/102658697-5d2c5700-4178-11eb-922e-170955166507.png">
](url)

**Comment by thohug** [comment]
@rogerfischer  or smoothing the line? I feel like this could be quite an issue for statistical purposes, as these lengths need to be reliable...

**Comment by djr4** [comment]
It would be possible to apply kalman filtering before doing distance
calculation, line would be smoother thus more accurate (At least that is my
intuition)

**Comment by rogerfischer** [comment]
The problem with smoothing is that you might have a straighter line, but it might be over buildings. 
I think the only way to make this perfect is to use a graph (=> OSMnx). 

Next week I will start with the train layer and then we can work from there. 

PS: @thohug How many walking examples do you have like this? 
      My walking is - if off - in an acceptable range for now. 
      
      PS: The best would be to use OSM to show the path you walked really and compare the distances.

**Comment by thohug** [comment]
@rogerfischer Ich versuche ein Bisschen darauf zu schauen. Ist mir aber schon ein paar Mal aufgefallen.

**Comment by rogerfischer** [comment]
@thohug Auf https://www.openstreetmap.org/directions#map=17/47.41088/8.54047&layers=T kann man die Wege ausmessen. Da können wir vergleichen, wie stark wir davon abweichen. 

<img width="1146" alt="Screen Shot 2020-12-21 at 2 53 48 PM" src="https://user-images.githubusercontent.com/7697124/102784226-60a02800-439c-11eb-8762-e2ca20ea0eb9.png">

PS: 
Im grossen Ganzen (ausser beim Zug) sollten die Abweichungen nicht zu gross sein.
Und sobald wir mit Graphen schaffen, wird es dann noch genauer. Aber das dauert noch ein wenig.

**Comment by djr4** [comment]
Note on this: We have barefoot map matcher but unless I only pick start and end point we have some strange paths. Like intersection forward and back if you remember.

**Comment by rogerfischer** [comment]
@djr4 Yes, the map matching algos are not ideal. There are other ways to obtain better results. E.g. multiple paths for a given segment.

**Comment by djr4** [comment]
By the way I'm testing right now some corrections that will make this much better with raw gps data, it does feel we are off by 150m or so depending on accumulation of points near static point. (noise)
